supertokens_python/recipe/multifactorauth/index.py

```diff
@@ -0,0 +1,146 @@
+    OTP_EMAIL="otp-email",
+    OTP_PHONE="otp-phone",
+    TOTP="totp",
+    OAUTH="oauth",
+)
+
+
+async def assert_allowed_to_setup_factor_else_throw_invalid_claim_error(
+    session: SessionContainer, factor_id: str, user_context: Dict[str, Any] = None
+) -> None:
+    ctx = get_user_context(user_context)
+
+    mfa_info = await update_and_get_mfa_related_info_in_session(
+        session=session, user_context=ctx
+    )
+    factors_setup_for_user = await get_factors_setup_for_user(
+        session.get_user_id(), ctx
+    )
+    await Recipe.get_instance().recipe_implementation.assert_allowed_to_setup_factor_else_throw_invalid_claim_error(
+        session=session,
+        factor_id=factor_id,
+        factors_setup_for_user=factors_setup_for_user,
+        mfa_requirements_for_auth=mfa_info.mfa_requirements_for_auth,
+        user_context=ctx,
+    )
+
+
+async def get_factors_setup_for_user(
+    user_id: str, user_context: Dict[str, Any] = None
+) -> Any:
+    ctx = get_user_context(user_context)
+    user = await get_user(user_id, ctx)
+    if user is None:
+        raise Exception("Unknown user id")
+
+    return await Recipe.get_instance().recipe_implementation.get_factors_setup_for_user(
+        user=user, user_context=ctx
+    )
+
+
+async def get_mfa_requirements_for_auth(
+    session: SessionContainer, user_context: Dict[str, Any] = None
+) -> Dict[str, Any]:
+    ctx = get_user_context(user_context)
+
+    mfa_info = await update_and_get_mfa_related_info_in_session(
+        session=session, user_context=ctx
+    )
+
+    return mfa_info["mfa_requirements_for_auth"]
+
+
+async def mark_factor_as_complete_in_session(
+    session: SessionContainer,
+    factor_id: str,
+    user_context: Dict[str, Any] | None = None,
+) -> None:
+    ctx = get_user_context(user_context)
+    await MultitenancyRecipe.get_instance().recipe_implementation.mark_factor_as_complete_in_session(
+        session=session, factor_id=factor_id, user_context=ctx
+    )
+
+
+async def get_factors_setup_for_user(
+    user_id: str, user_context: Dict[str, Any] = None
+) -> Any:
+    ctx = get_user_context(user_context)
+    try:
+        user = await get_user(user_id, ctx)
+    except SuperTokensUnknownUserIdError:
+        raise Exception("Unknown user id")
+
+    recipe_instance = await Recipe.get_instance()
+    return await recipe_instance.recipe_implementation.get_factors_setup_for_user(
+        {
+            "user": user,
+            "user_context": ctx,
+        }
+    )
+
+
+async def get_required_secondary_factors_for_user(
+    user_id: str, user_context: Dict[str, Any] = None
+) -> List[str]:
+    if user_context is None:
+        user_context = {}
+
+    recipe = MultiFactorAuth.get_instance()
+    tenant_id = await get_tenant_id(user_context)
+
+    return await recipe.recipe_implementation.get_required_secondary_factors_for_user(
+        {"userId": user_id, "tenantId": tenant_id, "userContext": user_context}
+    )
```

Comment:
this is an incorrect translation. The node code doesn't have tenant id being passed to getRequiredSecondaryFactorsForUser

Reply:
Removed for now